06-building-a-program-start-to-finish/writing-tests/output/invertedIndexSpec.js

```diff
@@ -0,0 +1,161 @@
+      const input = ['aa', 'aa', 'aa'];
+      const output = ['aa'];
+      expect(this.invertedIndex.uniqueWords(input))
+        .toEqual(output);
+    });
+  });
+
+  describe('Validate Files ', () => {
+    it('verifies that the JSON file is valid or invalid', () => {
+      expect(this.invertedIndex.validateFile(books)
+          .valid)
+        .toBe(true);
+      expect(this.invertedIndex.validateFile(smallfile)
+          .valid)
+        .toBe(true);
+      expect(this.invertedIndex.validateFile(emptyfile)
+          .valid)
+        .toBe(false);
+      expect(this.invertedIndex.validateFile(wrongdata)
+          .valid)
+        .toBe(false);
+    });
+  });
+
+  describe('CreateIndex', () => {
+    beforeAll(() => {
+      this.invertedIndex.createIndex(smallfile, ['alice', 'falls'], 'smallcorrectfile.json');
+      this.invertedIndex.createIndex(books, uniqueTermsBook, 'correct.json');
+    });
+    it('creates an index', () => {
+      expect(this.invertedIndex.getIndex('correct.json'))
+        .toBeTruthy();
+      expect(this.invertedIndex.getIndex('smallcorrectfile.json'))
+        .toBeTruthy();
+    });
+    it('creates the correct index', () => {
+      expect(this.invertedIndex.getIndex('correct.json')
+          .a)
+        .toEqual([true, true, true, true]);
+      expect(this.invertedIndex.getIndex('smallcorrectfile.json')
+          .falls)
+        .toEqual([true]);
+    });
+  });
+
+  describe('GetIndex', () => {
+    it('should return "undefined" if index does not exist', () => {
+      expect(this.invertedIndex.getIndex(' '))
+        .toEqual(undefined);
+      expect(this.invertedIndex.getIndex('fakeName'))
+        .toEqual(undefined);
+    });
+    it('returns the exact result of the index', () => {
+      this.invertedIndex.createIndex(smallfile, ['alice', 'falls', 'into', 'a', 'rabbit'], 'smallcorrectfile.json');
+      expect(this.invertedIndex.getIndex('smallcorrectfile.json'))
+        .toEqual({
+          a: [true],
+          alice: [true],
+          falls: [true],
+          into: [true],
+          rabbit: [true]
+        });
+    });
+  });
+
+  describe('searchIndex', () => {
+    it('returns empty if element being searched for does not exist', () => {
+      const keyword = 'alice';
+      const indexedData = {
+        dan: [false, false],
+        brown: [false, false],
+        john: [false, false],
+        grisham: [false, false]
+      };
+      const output = {};
+      expect(this.invertedIndex.searchIndex(keyword, indexedData))
+        .toEqual(output);
+    });
+
+    it('returns the element being searched for', () => {
+      const keyword = 'brown';
+      const indexedData = {
+        dan: [false, false],
+        brown: [false, false],
+        john: [false, false],
+        grisham: [false, false]
+      };
+      const output = {
+        brown: [false, false]
+      };
+      expect(this.invertedIndex.searchIndex(keyword, indexedData))
```

Comment:
How come you have to pass `indexedData` back to the `searchIndex` method?

Isn't it on the same object as the `createIndex` method?